modules/registry-configuring-storage-baremetal.adoc

```diff
@@ -40,30 +44,7 @@ $ oc get pod -n openshift-image-registry
 +
 [NOTE]
 =====
-* If the storage type is `emptyDIR`, the replica number cannot be greater than `1`.
-* If the storage type is `NFS`, you must enable the `no_wdelay` and `root_squash` mount options. For example:
-+
-[source,terminal]
-----
-# cat /etc/exports
-----
-+
-.Example output
-[source,terminal]
-----
-/mnt/data *(rw,sync,no_wdelay,root_squash,insecure,fsid=0)
-----
-+
-[source,terminal]
-----
-sh-4.2# exportfs -rv
-----
-+
-.Example output
-[source,terminal]
-----
-exporting *:/mnt/data
-----
+If the storage type is `emptyDIR`, the replica number cannot be greater than `1`.
```

Comment:
Seems for the `image-registry` deployment created by image registry cluster operator doesn't support `supplementalGroups`, so:
1. By default, if we use nfs storage as image registry backend storage, the replica number cannot be greater than `1`.
2. If the image registry cluster operator is unmanaged by CVO, and the `image-registry` deployment has the `supplementalGroups` config, the replica number can be greater than `1` for nfs storage.

Reply:
If use emptyDir, the replica number of image registry has to be 1. If not, it will have synchronization issue.

Reply:
@qinpingli The purpose of this change is to delete #1 because it references `nfs`. We want to remove references to `nfs` in bare metal and vSphere image registry config docs. So we should be fine leaving the `emptyDIR` note, IIUC.

Reply:
@bobfuru Yes, agree with keeping `emptyDIR` note. If we need to add some notes about `NFS`?

Reply:
@qinpingli For `modules/registry-configuring-storage-baremetal.adoc` and `modules/registry-configuring-storage-vsphere.adoc`, we are removing notes about `NFS` based on the findings by @huffmanca that "NFS can't be a recommended way of using image registry storage, as you would either have to leave the deployment/operator in an unmanaged state (not recommended) or leave the NFS export exposed (world readable/writable)."